bin/publish-stories.js

```diff
@@ -0,0 +1,65 @@
+const request = require('request');
+
+const existingComments = (accountName, accountKey, issueNumber) => new Promise((resolve, reject) => {
+
+    const commentsToDelete = [];
+    const auth = 'Basic ' + new Buffer(accountName + ':' + accountKey).toString('base64');
+    const options = {
+      url: `https://api.github.com/repos/newsuk/times-components/issues/${issueNumber}/comments`,
+      headers: {
+       'Authorization': auth,
+       'User-Agent': accountName
+      }
+    };
+
+    request.get(options, function(error, response, body) {
+        if (error) reject(error);
+        JSON.parse(body).forEach((comment) => {
+          if(comment.user.login === accountName) {
+            commentsToDelete.push(comment.id);
+          }
+        })
+        resolve(commentsToDelete);
+      });
+});
+
+const deleteComments = (commentsToDelete, accountName, accountKey) => new Promise((resolve, reject) => {
+  commentsToDelete.forEach((commentId) => {
+    const auth = 'Basic ' + new Buffer(accountName + ':' + accountKey).toString('base64');
+    const deleteOptions = {
+      url: `https://api.github.com/repos/newsuk/times-components/issues/comments/${commentId}`,
+      headers: {
+        'Authorization': auth,
+        'User-Agent': accountName
+      }
+    };
+
+    request.delete(deleteOptions, function(error, response, body) {
+      if (error) reject(error);
+      resolve('deleted');
```

Comment:
that looks wierd:
this is called foreach commentId but you resolve the same promise.
Instead you should use I'd suggest you  rewrite it as `Promise.all( commentsToDelete.map( ... ) )`